File: power.py
@schema 
class Power(dj.Computed):
    definition = """  # LFP power in each frequency band
    -> LFP
    -> FrequencyBand
    -----
    power :float  # mV^2 Hz -- averge power in the band
    """

    def _make_tuples(self, key):
        # fetch required data
        lo, hi = (FrequencyBand() & key).fetch1['freq_lo', 'freq_hi']
        signal, dt = (LFP() & key).fetch1['voltage', 'dt']
        # compute
        signal = band_pass_filter(signal, lo, hi, dt)
        power = compute_average_power(signal, dt)
        # submit 
        self.insert1(dict(key, power=power))

File: Power.m
%{
example.Power (computed)  # LFP power in each frequency band
-> example.LFP
-> example.FrequencyBand
-----
power :float  # mV^2 Hz -- average power in the band
%}

classdef Power < dj.Relvar & dj.AutoPopulate
    methods(Access=protected)
        function makeTuples(self, key)
            % fetch required data 
            [lo, hi] = fetch1(example.FrequencyBand & key, 'freq_lo', 'freq_hi');
            [signal, dt] = fetch1(example.LFP & key, 'voltage', 'dt');
            % compute
            signal = band_pass_filter(signal, lo, hi, dt);
            key.power = compute_average_power(signal, dt);
            % submit
            self.insert(key)
        end
    end
end
